examples/speculative/speculative.cpp

```diff
@@ -219,22 +224,24 @@ int main(int argc, char ** argv) {
                     float p_tgt = 0, p_dft = 0;
 
                     // GGML_ASSERT(dist_tgt.size() == dist_dft.size());
-                    for (int s = 0; s < n_seq_dft; ++s) {
-                        if (!drafts[s].active) {
-                            continue;
-                        }
+
+                    while (active_seqs.size() > 0) {
+                        // randomly select a sequence to verify from active sequences
+                        int s = *std::next(active_seqs.begin(), u_int_dist(rng) % active_seqs.size());
```

Comment:
For this to be a statistically accurate selection, you should initialize a new distribution based on active_seqs.size() when needed, and skip the modulo.

Reply:
thanks for pointing this out 😄 fixed in [[7463569](https://github.com/ggerganov/llama.cpp/pull/5625/commits/7463569cad717b78fe5c62508636f01a706757ce)]